src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -974,6 +977,12 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                 // Any errors generated below here are considered Emit diagnostics 
                 // and will not be reported to callers Compilation.GetDiagnostics()
 
+                ImmutableArray<SourceSpan> dynamicAnalysisSpans = ImmutableArray<SourceSpan>.Empty;
+                if (_moduleBeingBuiltOpt.EmitOptions.EmitDynamicAnalysisData)
+                {
+                    flowAnalyzedBody = Instrumentation.InjectInstrumentation(methodSymbol, flowAnalyzedBody, methodOrdinal, compilationState, _compilation, _diagnostics, _debugDocumentProvider, out dynamicAnalysisSpans);
```

Comment:
So far I haven't seen anything that would prevent us from doing instrumentation as part of the lowering pass. The location information should be accurate, after all this is were we add sequence points.